x-pack/plugins/watcher/public/models/action/webhook.action.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { get } from 'lodash';
+import { BaseAction } from './base_action';
+import { i18n } from '@kbn/i18n';
+
+const requiredFields = ['host', 'port'];
+const optionalFields = [
+  'scheme',
+  'path',
+  'method',
+  'headers',
+  'params',
+  'auth',
+  'body',
+  'proxy',
+  'connection_timeout',
+  'read_timeout',
+  'url'
+];
+
+const allFields = [...requiredFields, ...optionalFields];
+
+export class WebhookAction extends BaseAction {
+  constructor(props = {}) {
+    super(props);
+
+    allFields.forEach((field) => {
+      this[field] = get(props, field);
+    });
+
+    this.fullPath = this.url ? this.url : this.host + this.port + this.path;
+  }
+
+  get upstreamJson() {
+    // Add all required fields to the request body
+    let result = requiredFields.reduce((acc, field) => {
+      acc[field] = this[field];
+      return acc;
+    }, super.upstreamJson);
+
+    // If optional fields have been set, add them to the body
+    result = optionalFields.reduce((acc, field) => {
+      if (this[field]) {
+        acc[field] = this[field];
+      }
+      return acc;
+    }, result);
+
+    return result;
+  }
+
+  get description() {
+    return i18n.translate('xpack.watcher.models.webhookAction.description', {
+      defaultMessage: 'Webhook will trigger a {method} request on {fullPath}',
+      values: {
+        method: this.method,
+        fullPath: this.fullPath
+      }
+    });
+  }
+
+  get simulateMessage() {
+    return i18n.translate('xpack.watcher.models.webhookAction.simulateMessage', {
+      defaultMessage: 'Sample request sent to {fullPath}',
+      values: {
+        fullPath: this.fullPath
+      }
+    });
+  }
+
+  get simulateFailMessage() {
+    return i18n.translate('xpack.watcher.models.webhookAction.simulateFailMessage', {
+      defaultMessage: 'Failed to send request to {fullPath}.',
+      values: {
+        fullPath: this.fullPath
+      }
+    });
+  }
+
+  static fromUpstreamJson(upstreamAction) {
+    return new WebhookAction(upstreamAction);
+  }
+
+  /**
+   * NOTE:
+   *
+   * I don't seem to find in the UI where those static properties are actuall used.
+   * It looks like we used to have a UI to create an action and that currently we only have
+   * the "advanced watcher" creation through the JSON editor.
+   *
+   * ---> ./components/watch_actions/components/watch_action/watch_actions.html
+   * is where it seems that this is read. But I can't access that component navigatint the UI
+   *
+   */
+  // static typeName = i18n.translate('xpack.watcher.models.webhookAction.typeName', {
```

Comment:
Is this commented-out code needed?

Reply:
I was hoping the reviewer(s) to be able to answer this commented code. 😊 Because it has been copied from other watcher action but those static variables don't seem to be used anywhere in the UI. 
@bmcconaghy do you know if in the past there was a UI form or template to create the actions? (instead of the advanced JSON editor that we have now)

Reply:
I am not familiar with those comments at all unfortunately

Reply:
Ok thanks @chrisronline